datalad/distribution/get.py

```diff
@@ -49,53 +51,209 @@
     the directory are obtained. Recursion into subdatasets is supported too. If
-    enabled, potential subdatasets are detected and installed sequentially, in
-    order to fulfill a request. However, this implicit installation of
-    subdatasets is done only, if an explicitly specified path belongs to such a
-    subdataset and for that purpose `recursion_limit` is ignored. Otherwise get
-    recurses into already installed subdatasets only and the depth of this
-    recursion can be limited by `recursion_limit`.
+    enabled, relevant subdatasets are detected and installed in order to
+    fulfill a request.
+
+    Known data locations for each requested file are evaluated and data are
+    obtained from the best/fastest/cheapest location, unless a specific
+    source is identified.
 
     .. note::
       Power-user info: This command used :command:`git annex get` to fulfill
-      requests. Subdatasets are obtained via the :func:`~datalad.api.install`
-      command.
+      file handles.
     """
 
     _params_ = dict(
         dataset=Parameter(
             args=("-d", "--dataset"),
             metavar="PATH",
-            doc="""specify the dataset to perform the add operation on.  If
-            no dataset is given, an attempt is made to identify the dataset
-            based on the current working directory and/or the `path` given""",
+            doc="""specify the dataset to perform the add operation on, in
+            which case `path` arguments are interpreted as being relative
+            to this dataset.  If no dataset is given, an attempt is made to
+            identify a dataset for each input `path`""",
             constraints=EnsureDataset() | EnsureNone()),
         path=Parameter(
             args=("path",),
             metavar="PATH",
             doc="""path/name of the requested dataset component. The component
-            must already be known to the dataset.""",
+            must already be known to a dataset. To add new components to a
+            dataset use the `add` command""",
             nargs="*",
             constraints=EnsureStr() | EnsureNone()),
         source=Parameter(
             args=("-s", "--source",),
             metavar="LABEL",
-            doc="""label of the data source to be used to fulfill the request.
+            doc="""label of the data source to be used to fulfill requests.
             This can be the name of a dataset :term:`sibling` or another known
             source""",
             constraints=EnsureStr() | EnsureNone()),
         recursive=recursion_flag,
         recursion_limit=recursion_limit,
+        fulfill_datasets=Parameter(
+            args=("--fulfill-datasets",),
+            action='store_true',
+            doc="""whether to fulfill file handles in obtained datasets. If
+            true, then a subdataset given via `path` will not only be
+            installed, but also all of its file handles will be fulfilled."""),
```

Comment:
Why? They either state intension to do the same thing (answer yes) or different (answer no)